app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt

```diff
@@ -30,29 +30,73 @@ import com.wire.android.ui.common.WireDialog
-                onClick = hideDialogStatus,
-                text = stringResource(id = R.string.label_ok),
-                type = WireDialogButtonType.Primary,
-            )
+    val text: String = stringResource(id = if (isFileSharingEnabled) R.string.sharing_files_enabled else R.string.sharing_files_disabled)
+
+    WireDialog(
+        title = stringResource(id = R.string.team_settings_changed),
+        text = text,
+        onDismiss = hideDialogStatus,
+        optionButton1Properties = WireDialogButtonProperties(
+            onClick = hideDialogStatus,
+            text = stringResource(id = R.string.label_ok),
+            type = WireDialogButtonType.Primary,
         )
+    )
+}
+
+@Composable
+fun SelfDeletingMessagesDialog(
+    isSelfDeletingMessagesEnabled: Boolean,
+    enforcedTimeout: Int?,
+    hideDialogStatus: () -> Unit,
+) {
+    val text: String = if (isSelfDeletingMessagesEnabled) {
+        enforcedTimeout?.let { rawTimeout ->
+            val formattedTimeout = formatEnforcedTimeout(rawTimeout)
+            stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout, formattedTimeout)
+        } ?: stringResource(id = R.string.self_deleting_messages_team_setting_enabled)
+    } else stringResource(id = R.string.self_deleting_messages_team_setting_disabled)
+
+
+    WireDialog(
+        title = stringResource(id = R.string.team_settings_changed),
+        text = text,
+        onDismiss = hideDialogStatus,
+        optionButton1Properties = WireDialogButtonProperties(
+            onClick = hideDialogStatus,
+            text = stringResource(id = R.string.label_ok),
+            type = WireDialogButtonType.Primary,
+        )
+    )
+}
+
+@Composable
+private fun formatEnforcedTimeout(rawTimeoutInSeconds: Int): String {
+    val timeout10Seconds = 10.seconds
+    val timeout5Minutes = 300.seconds
+    val timeout1Hour = 3600.seconds
+    val timeout1Day = 86400.seconds
+    val timeout1Week = 604800.seconds
+    val timeout4Weeks = 2419200.seconds
+
+    return when (rawTimeoutInSeconds.toDuration(DurationUnit.SECONDS)) {
+        timeout10Seconds -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_seconds)
+        timeout5Minutes -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_minutes)
+        timeout1Hour -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_one_hour)
+        timeout1Day -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_one_day)
+        timeout1Week -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_one_week)
+        timeout4Weeks -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_four_weeks)
+        else -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled)
     }
 }
```

Comment:
All this is already implemented in SelfDeletionDuration, however now I see I forgot to make it work with localisations, so it needs a small refactor